File: android/src/main/java/com/senziio/capacitorsse/SenziioSSEPlugin.java
package com.senziio.capacitorsse;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SenziioSSE")
public class SenziioSSEPlugin extends Plugin {

    private SenziioSSE sse;

    @Override
    public void load() {
        super.load();
        this.sse = new SenziioSSE(getContext());
    }

    @PluginMethod(returnType = PluginMethod.RETURN_CALLBACK)
    public void connect(PluginCall call) {
        try {
            String url = call.getString("url");

            if (url == null) {
                throw new IllegalArgumentException("URL es requerida");
            }

            call.setKeepAlive(true);
            sse.connect(url, new SenziioSSEPluginCallback(call, bridge));
        } catch (Exception e) {
            call.reject(e.getMessage(), e);
        }
    }

    @PluginMethod
    public void disconnect(PluginCall call) {
        try {
            String connectionId = call.getString("connectionId");

            if (connectionId == null) {
                throw new IllegalArgumentException("connectionId es requerido");
            }

            sse.disconnect(connectionId);
            call.resolve();
        } catch (Exception e) {
            call.reject(e.getMessage(), e);
        }
    }
}

File: ios/Sources/SenziioSSEPlugin/SenziioSSEPlugin.swift
import Foundation
import Capacitor

@objc(SenziioSSEPlugin)
class SenziioSSEPlugin: CAPPlugin, CAPBridgedPlugin {

    public let identifier = "SenziioSSEPlugin"
    public let jsName = "SenziioSSE"

    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "connect", returnType: CAPPluginReturnCallback),
        CAPPluginMethod(name: "disconnect", returnType: CAPPluginReturnPromise)
    ]

    private let implementation = SenziioSSE()

    @objc func connect(_ call: CAPPluginCall) {
        guard let urlString = call.getString("url") else {
            call.reject("URL is required")
            return
        }

        guard let url = URL(string: urlString) else {
            call.reject("Invalid URL format")
            return
        }

        call.keepAlive = true
        implementation.connect(url, SenziioSSEPluginCallback(call, bridge))
    }

    @objc func disconnect(_ call: CAPPluginCall) {
        guard let connectionId = call.getString("connectionId") else {
            call.reject("connectionId is required")
            return;
        }

        implementation.disconnect(connectionId)
        call.resolve()
    }
}
